src/main/java/pib/tasks/Deadline.java

```diff
@@ -0,0 +1,90 @@
+        super(description, printMessage);
+        this.date = date;
+        this.time = time;
+    }
+
+    private Deadline(String description, int isDone, String date, String time, boolean printMessage) {
+        super(description, isDone, printMessage);
+        this.date = date;
+        this.time = time;
+    }
+
+    /**
+     * A public factory method to create a Deadline task
+     *
+     * @param details String containing the description, date and time
+     * @param printMessage Boolean to indicate whether to print the success message after each Task is added
+     * @return Deadline object with description initialised and isDone set to 0
+     * @throws PibException when user inputs blank task description
+     * @throws PibException when user inputs wrongly formatted command to create a new Deadline
+     * @throws PibException when user inputs wrongly formatted date/time
+     */
+    public static Deadline createDeadline(String details, boolean printMessage) throws PibException {
+        try {
+            int byIndex = details.indexOf("/by ");
+            String description = details.substring(0, byIndex).trim();
+            if (description.isBlank()) {
+                throw new PibException("empty-task-description");
+            }
+            String[] dateTime = details.substring(byIndex + 4).trim().split(" ");
+            String date = LocalDate.parse(dateTime[0].trim()).format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+            String time = LocalTime.parse(dateTime[1].trim(), DateTimeFormatter.ofPattern("HHmm")).format(DateTimeFormatter.ofPattern("hh:mm a"));
+            return new Deadline(description, date, time, printMessage);
+        } catch (IndexOutOfBoundsException e) {
+            throw new PibException("d-wrong-format");
+        } catch (DateTimeParseException e) {
+            throw new PibException("wrong-datetime-format");
+        }
+    }
+
+    /**
+     * A public factory method to create a Deadline task
+     *
+     * @param description task description
+     * @param isDone value 0 (false) or 1 (true)
+     * @param date String showing date of Deadline task
+     * @param time String showing time of Deadline task
+     * @param printMessage Boolean to indicate whether to print the success message after each Task is added
+     * @return Deadline object with these 4 fields initialised
+     */
+    public static Deadline createDeadline(String description, int isDone, String date, String time, boolean printMessage) {
+        return new Deadline(description, isDone, date, time, printMessage);
+    }
+
+    /**
+     * Public method to convert task to a string format used to save inside a .txt file
+     *
+     * @return string format of Deadline task to be saved
+     */
+    public String toDataString() {
+        return "D," + getIsDone() + "," + getDescription() + "," + date + "," + time + System.lineSeparator();
+    }
```

Comment:
Perhaps you can improve readability here by using String.format()? 
Noticed the same issues in several other places that returns a task as a String.